src/sentry/workflow_engine/migration_helpers/rule_action.py

```diff
@@ -0,0 +1,136 @@
+
+    :param actions: list of action data (Rule.data.actions)
+    :return: list of notification actions (Action)
+    """
+
+    notification_actions: list[Action] = []
+
+    for action in actions:
+        # Use Rule.integration.provider to get the action type
+        action_type = RULE_REGISTRY_ID_2_INTEGRATION_PROVIDER.get(action["id"])
+        if action_type is None:
+            _logger.warning(
+                "Action type not found for action",
+                extra={
+                    "action_id": action["id"],
+                    "action_uuid": action["uuid"],
+                },
+            )
+            continue
+
+        # For all integrations, the target type is specific
+        # For email, the target type is user
+        # For sentry app, the target type is sentry app
+        # FWIW, we don't use target type for issue alerts
+        target_type = ACTION_TYPE_2_TARGET_TYPE_RULE_REGISTRY.get(action_type)
+
+        if target_type == ActionTarget.SPECIFIC:
+
+            # Get the integration_id
+            integration_id_key = ACTION_TYPE_2_INTEGRATION_ID_KEY.get(action_type)
+            if integration_id_key is None:
+                # we should always have an integration id key if target type is specific
+                # TODO(iamrajjoshi): Should we fail loudly here?
+                _logger.warning(
+                    "Integration ID key not found for action type",
+                    extra={
+                        "action_type": action_type,
+                        "action_id": action["id"],
+                        "action_uuid": action["uuid"],
+                    },
+                )
+                continue
+            integration_id = action.get(integration_id_key)
+
+            # Get the target_identifier if it exists
+            target_identifier_key = ACTION_TYPE_2_TARGET_IDENTIFIER_KEY.get(action_type)
+            if target_identifier_key is not None:
+                target_identifier = action.get(target_identifier_key)
+
+            # Get the target_display if it exists
+            target_display_key = ACTION_TYPE_2_TARGET_DISPLAY_KEY.get(action_type)
+            if target_display_key is not None:
+                target_display = action.get(target_display_key)
+
+        notification_action = Action(
+            type=action_type,
+            data=(
+                # If the target type is specific, sanitize the action data
+                # Otherwise, use the action data as is
+                sanitize_to_action(action, action_type)
+                if target_type == ActionTarget.SPECIFIC
+                else action
+            ),
+            integration_id=integration_id,
+            target_identifier=target_identifier,
+            target_display=target_display,
+            target_type=target_type,
+        )
+
+        notification_actions.append(notification_action)
+
+    # Bulk create the actions, note: this won't call save(), not sure if we need to
```

Comment:
this is probably fine unless you have pre and/or post save stuff you need to run